app/services/migration/migrators/user.rb

```diff
@@ -34,34 +34,40 @@ def call
       migrate(self.class.ecf_users) do |ecf_user|
         user = ::Migration::Users::Creator.new(ecf_user).find_or_initialize
 
-        application_trns = validated_application_trns(ecf_user)
-        raise_if_multiple_application_trns_and_no_teacher_profile_trn!(application_trns, ecf_user)
-        ecf_trn = ecf_user.teacher_profile&.trn || application_trns.last
+        application_trns_by_verified = application_trns_by_verified(ecf_user)
+
+        raise_if_multiple_verified_application_trns_and_no_teacher_profile_trn!(application_trns_by_verified[true], ecf_user)
+        ecf_verified_trn = ecf_user.teacher_profile&.trn || application_trns_by_verified[true]&.first
+        ecf_unverified_trn = application_trns_by_verified[false]&.first
 
         npq_application = most_recent_created_npq_application(ecf_user)
         email = npq_application&.participant_identity&.email
 
         user.update!(
-          trn: ecf_trn || user.trn,
+          trn: ecf_verified_trn || ecf_unverified_trn || user.trn,
           full_name: ecf_user.full_name || user.full_name,
           email:,
           uid: ecf_user.get_an_identity_id || user.uid,
           date_of_birth: npq_application.date_of_birth || user.date_of_birth,
           national_insurance_number: npq_application.nino || user.national_insurance_number,
           active_alert: npq_application.active_alert || user.active_alert,
-          trn_verified: ecf_trn.present? || user&.trn_verified,
+          trn_verified: ecf_verified_trn || (!ecf_unverified_trn && user&.trn_verified),
```

Comment:
Got confused here in this line..
Maybe if we do something like this to make it easier to read please?

`trn_verified: ecf_verified_trn.present? || (ecf_unverified_trn.blank? && user&.trn_verified)`